### PR TITLE
Handle parsing of Image metadata results better

### LINKED ISSUE #110: CLI fails to parse response from Engine when getting image manifest
After adding alpine:latest for analysis, retrieving it's manifest metadata fails unless you use the json flag

```zsh
[anchore@anchore-cli-7cf49bf7d-58cnc anchore-cli]$ anchore-cli --debug image metadata alpine:latest manifest
DEBUG:anchorecli.clients.apiexternal:As Account = None
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localdev-anchore-engine-api.default.svc.cluster.local:8228
DEBUG:urllib3.connectionpool:http://localdev-anchore-engine-api.default.svc.cluster.local:8228 "GET /v1/ HTTP/1.1" 200 5
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localdev-anchore-engine-api.default.svc.cluster.local:8228
DEBUG:urllib3.connectionpool:http://localdev-anchore-engine-api.default.svc.cluster.local:8228 "GET /swagger.json HTTP/1.1" 200 187677
DEBUG:anchorecli.clients.apiexternal:As Account = None
DEBUG:anchorecli.clients.apiexternal:GET url=http://localdev-anchore-engine-api.default.svc.cluster.local:8228/v1/images
DEBUG:anchorecli.clients.apiexternal:GET params={'fulltag': 'alpine:latest', 'history': 'false'}
DEBUG:anchorecli.clients.apiexternal:Use get body because detected api version (0, 1, 15) < (0, 1, 6)? False
DEBUG:anchorecli.clients.apiexternal:GET insecure=True
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localdev-anchore-engine-api.default.svc.cluster.local:8228
DEBUG:urllib3.connectionpool:http://localdev-anchore-engine-api.default.svc.cluster.local:8228 "GET /v1/images?fulltag=alpine%3Alatest&history=false HTTP/1.1" 200 1621
DEBUG:anchorecli.cli.image:discovery from input: tag : alpine:latest : sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65
DEBUG:anchorecli.clients.apiexternal:As Account = None
DEBUG:anchorecli.clients.apiexternal:GET url=http://localdev-anchore-engine-api.default.svc.cluster.local:8228/v1/images/sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65/metadata/manifest
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localdev-anchore-engine-api.default.svc.cluster.local:8228
DEBUG:urllib3.connectionpool:http://localdev-anchore-engine-api.default.svc.cluster.local:8228 "GET /v1/images/sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65/metadata/manifest HTTP/1.1" 200 756
DEBUG:anchorecli.cli.utils:fetched httpcode from response: 200
Error: Expecting value: line 1 column 1 (char 0)
```

Works when you add the json flag though:

```zsh
[anchore@anchore-cli-7cf49bf7d-58cnc anchore-cli]$ anchore-cli --json image metadata alpine:latest manifest
{
    "imageDigest": "sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65",
    "metadata": "eyJzY2hlbWFWZXJzaW9uIjogMiwgIm1lZGlhVHlwZSI6ICJhcHBsaWNhdGlvbi92bmQuZG9ja2Vy\nLmRpc3RyaWJ1dGlvbi5tYW5pZmVzdC52Mitqc29uIiwgImNvbmZpZyI6IHsibWVkaWFUeXBlIjog\nImFwcGxpY2F0aW9uL3ZuZC5kb2NrZXIuY29udGFpbmVyLmltYWdlLnYxK2pzb24iLCAic2l6ZSI6\nIDE1MDksICJkaWdlc3QiOiAic2hhMjU2OmEyNGJiNDAxMzI5NmY2MWU4OWJhNTcwMDVhN2IzZTUy\nMjc0ZDhlZGQzYWUyMDc3ZDA0Mzk1ZjgwNmI2M2Q4M2UifSwgImxheWVycyI6IFt7Im1lZGlhVHlw\nZSI6ICJhcHBsaWNhdGlvbi92bmQuZG9ja2VyLmltYWdlLnJvb3Rmcy5kaWZmLnRhci5nemlwIiwg\nInNpemUiOiAyNzk3NTQxLCAiZGlnZXN0IjogInNoYTI1NjpkZjIwZmE5MzUxYTE1NzgyYzY0ZTZk\nZGRiMmQ0YTZmNTBiZjZkMzY4ODA2MGEzNGM0MDE0YjBkOWE3NTJlYjRjIn1dfQ==\n",
    "metadata_type": "manifest"
}
```

In [format_content_query](https://github.com/anchore/anchore-cli/blob/master/anchorecli/cli/utils.py#L338) it returns an empty string, because the normal manifest metadata response looks like this, and doesn't have a "content" field: 
```json
{
    "imageDigest": "sha256:156f59dc1cbe233827642e09ed06e259ef6fa1ca9b2e29d52ae14d5e7b79d7f0",
    "metadata": "",
    "metadata_type": "manifest"
}
```

When the [CLI method](https://github.com/anchore/anchore-cli/blob/master/anchorecli/cli/image.py#L277) for querying metadata sees that we're trying to get a manifest, it tries to load it as JSON, but it's an empty string that will end up getting returned from format_output: 
```python
                    if metadata_type in ['manifest', 'docker_history']:
                        o = json.loads(anchorecli.cli.utils.format_output(config, 'image_metadata', {'query_type':metadata_type}, ret['payload']))
                        print(json.dumps(o, indent=4))
```

### PR DESCRIPTION
**Pull request by dakaneye**
Fixes #110 
Signed-off-by: Samuel Dacanay <sam.dacanay@anchore.com>

Local Tests:

Pushed the ruby:latest image into my anchore-engine for analysis prior to running this. 
Local deployment: helm w/ minikube
```zsh
samdacanay@SDACANAY anchore-helm-charts %     kubectl run -i --tty anchore-cli --restart=Always --image anchore-cli:dev  --env ANCHORE_CLI_USER=admin --env ANCHORE_CLI_PASS=${ANCHORE_CLI_PASS} --env ANCHORE_CLI_URL=http://localdev-anchore-engine-api.default.svc.cluster.local:8228/v1/
kubectl run --generator=deployment/apps.v1 is DEPRECATED and will be removed in a future version. Use kubectl run --generator=run-pod/v1 or kubectl create instead.
If you don't see a command prompt, try pressing enter.
[anchore@anchore-cli-59b7667b76-c6z8l anchore-cli]$ anchore-cli image metadata ruby:latest manifest
Image Digest: sha256:bcb6b718fd2709e6cf49d8cde41da3e10da4e14b0f1f28c33fd23f04073745a3
Metadata: {"schemaVersion": 2, "mediaType": "application/vnd.docker.distribution.manifest.v2+json", "config": {"mediaType": "application/vnd.docker.container.image.v1+json", "size": 7312, "digest": "sha256:958d3491c09ae500375011be3ee77aec02387e11ba85d486fe365e667bd5e29a"}, "layers": [{"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 50396000, "digest": "sha256:d6ff36c9ec4822c9ff8953560f7ba41653b348a9c1136755e653575f58fbded7"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 7811570, "digest": "sha256:c958d65b3090aefea91284d018b2a86530a3c8174b72616c4e76993c696a5797"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 9996337, "digest": "sha256:edaf0a6b092f5673ec05b40edb606ce58881b2f40494251117d31805225ef064"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 51829826, "digest": "sha256:80931cf6881673fd161a3fd73e8971fe4a569fd7fbb44e956d261ca58d97dfab"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 192243499, "digest": "sha256:bc1b8aca3825e6c3b57e28c8075180f97879161dd58333de13f6b6da0f842af3"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 199, "digest": "sha256:e64edafe3f358fb6f209f140af24d98362473617360d6e5141a0edf697a823bb"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 22878106, "digest": "sha256:5f7e31a225bcd7cfd4c2935ad7964299465712f2f205679d3765484fdd8cf2eb"}, {"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "size": 143, "digest": "sha256:a7a2582e3a281ffbecf7e71382dee50443dfc4753756680f35b5be99910c6443"}]}
Metadata Type: manifest

[anchore@anchore-cli-59b7667b76-c6z8l anchore-cli]$ anchore-cli image metadata ruby:latest dockerfile
Image Digest: sha256:bcb6b718fd2709e6cf49d8cde41da3e10da4e14b0f1f28c33fd23f04073745a3
Metadata Type: dockerfile

[anchore@anchore-cli-59b7667b76-c6z8l anchore-cli]$ anchore-cli image metadata ruby:latest docker_history
Image Digest: sha256:bcb6b718fd2709e6cf49d8cde41da3e10da4e14b0f1f28c33fd23f04073745a3
Metadata: [{"Comment": "", "Created": "2020-08-04T15:42:33.850643052Z", "CreatedBy": "/bin/sh -c #(nop) ADD file:4b03b5f551e3fbdf47ec609712007327828f7530cc3455c43bbcdcaf449a75a9 in / ", "Id": "sha256:d6ff36c9ec4822c9ff8953560f7ba41653b348a9c1136755e653575f58fbded7", "Size": 50396000, "Tags": []}, {"Comment": "", "Created": "2020-08-04T15:42:34.128304606Z", "CreatedBy": "/bin/sh -c #(nop)  CMD [\"bash\"]", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-04T23:26:28.651181885Z", "CreatedBy": "/bin/sh -c apt-get update && apt-get install -y --no-install-recommends \t\tca-certificates \t\tcurl \t\tnetbase \t\twget \t&& rm -rf /var/lib/apt/lists/*", "Id": "sha256:c958d65b3090aefea91284d018b2a86530a3c8174b72616c4e76993c696a5797", "Size": 7811570, "Tags": []}, {"Comment": "", "Created": "2020-08-04T23:26:34.65752821Z", "CreatedBy": "/bin/sh -c set -ex; \tif ! command -v gpg > /dev/null; then \t\tapt-get update; \t\tapt-get install -y --no-install-recommends \t\t\tgnupg \t\t\tdirmngr \t\t; \t\trm -rf /var/lib/apt/lists/*; \tfi", "Id": "sha256:edaf0a6b092f5673ec05b40edb606ce58881b2f40494251117d31805225ef064", "Size": 9996337, "Tags": []}, {"Comment": "", "Created": "2020-08-04T23:26:55.098815484Z", "CreatedBy": "/bin/sh -c apt-get update && apt-get install -y --no-install-recommends \t\tgit \t\tmercurial \t\topenssh-client \t\tsubversion \t\t\t\tprocps \t&& rm -rf /var/lib/apt/lists/*", "Id": "sha256:80931cf6881673fd161a3fd73e8971fe4a569fd7fbb44e956d261ca58d97dfab", "Size": 51829826, "Tags": []}, {"Comment": "", "Created": "2020-08-04T23:27:46.998510292Z", "CreatedBy": "/bin/sh -c set -ex; \tapt-get update; \tDEBIAN_FRONTEND=noninteractive \tapt-get install -y --no-install-recommends \t\tautoconf \t\tautomake \t\tbzip2 \t\tdpkg-dev \t\tfile \t\tg++ \t\tgcc \t\timagemagick \t\tlibbz2-dev \t\tlibc6-dev \t\tlibcurl4-openssl-dev \t\tlibdb-dev \t\tlibevent-dev \t\tlibffi-dev \t\tlibgdbm-dev \t\tlibglib2.0-dev \t\tlibgmp-dev \t\tlibjpeg-dev \t\tlibkrb5-dev \t\tliblzma-dev \t\tlibmagickcore-dev \t\tlibmagickwand-dev \t\tlibmaxminddb-dev \t\tlibncurses5-dev \t\tlibncursesw5-dev \t\tlibpng-dev \t\tlibpq-dev \t\tlibreadline-dev \t\tlibsqlite3-dev \t\tlibssl-dev \t\tlibtool \t\tlibwebp-dev \t\tlibxml2-dev \t\tlibxslt-dev \t\tlibyaml-dev \t\tmake \t\tpatch \t\tunzip \t\txz-utils \t\tzlib1g-dev \t\t\t\t$( \t\t\tif apt-cache show 'default-libmysqlclient-dev' 2>/dev/null | grep -q '^Version:'; then \t\t\t\techo 'default-libmysqlclient-dev'; \t\t\telse \t\t\t\techo 'libmysqlclient-dev'; \t\t\tfi \t\t) \t; \trm -rf /var/lib/apt/lists/*", "Id": "sha256:bc1b8aca3825e6c3b57e28c8075180f97879161dd58333de13f6b6da0f842af3", "Size": 192243499, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:13:58.595202642Z", "CreatedBy": "/bin/sh -c set -eux; \tmkdir -p /usr/local/etc; \t{ \t\techo 'install: --no-document'; \t\techo 'update: --no-document'; \t} >> /usr/local/etc/gemrc", "Id": "sha256:e64edafe3f358fb6f209f140af24d98362473617360d6e5141a0edf697a823bb", "Size": 199, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:13:58.782438185Z", "CreatedBy": "/bin/sh -c #(nop)  ENV LANG=C.UTF-8", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:13:58.962023988Z", "CreatedBy": "/bin/sh -c #(nop)  ENV RUBY_MAJOR=2.7", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:13:59.130932994Z", "CreatedBy": "/bin/sh -c #(nop)  ENV RUBY_VERSION=2.7.1", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:13:59.306131029Z", "CreatedBy": "/bin/sh -c #(nop)  ENV RUBY_DOWNLOAD_SHA256=b224f9844646cc92765df8288a46838511c1cec5b550d8874bd4686a904fcee7", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:16:44.796694796Z", "CreatedBy": "/bin/sh -c set -eux; \t\tsavedAptMark=\"$(apt-mark showmanual)\"; \tapt-get update; \tapt-get install -y --no-install-recommends \t\tbison \t\tdpkg-dev \t\tlibgdbm-dev \t\truby \t; \trm -rf /var/lib/apt/lists/*; \t\twget -O ruby.tar.xz \"https://cache.ruby-lang.org/pub/ruby/${RUBY_MAJOR%-rc}/ruby-$RUBY_VERSION.tar.xz\"; \techo \"$RUBY_DOWNLOAD_SHA256 *ruby.tar.xz\" | sha256sum --check --strict; \t\tmkdir -p /usr/src/ruby; \ttar -xJf ruby.tar.xz -C /usr/src/ruby --strip-components=1; \trm ruby.tar.xz; \t\tcd /usr/src/ruby; \t\t{ \t\techo '#define ENABLE_PATH_CHECK 0'; \t\techo; \t\tcat file.c; \t} > file.c.new; \tmv file.c.new file.c; \t\tautoconf; \tgnuArch=\"$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)\"; \t./configure \t\t--build=\"$gnuArch\" \t\t--disable-install-doc \t\t--enable-shared \t; \tmake -j \"$(nproc)\"; \tmake install; \t\tapt-mark auto '.*' > /dev/null; \tapt-mark manual $savedAptMark > /dev/null; \tfind /usr/local -type f -executable -not \\( -name '*tkinter*' \\) -exec ldd '{}' ';' \t\t| awk '/=>/ { print $(NF-1) }' \t\t| sort -u \t\t| xargs -r dpkg-query --search \t\t| cut -d: -f1 \t\t| sort -u \t\t| xargs -r apt-mark manual \t; \tapt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \t\tcd /; \trm -r /usr/src/ruby; \t! dpkg -l | grep -i ruby; \t[ \"$(command -v ruby)\" = '/usr/local/bin/ruby' ]; \truby --version; \tgem --version; \tbundle --version", "Id": "sha256:5f7e31a225bcd7cfd4c2935ad7964299465712f2f205679d3765484fdd8cf2eb", "Size": 22878106, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:16:45.013175781Z", "CreatedBy": "/bin/sh -c #(nop)  ENV GEM_HOME=/usr/local/bundle", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:16:45.182385072Z", "CreatedBy": "/bin/sh -c #(nop)  ENV BUNDLE_SILENCE_ROOT_WARNING=1 BUNDLE_APP_CONFIG=/usr/local/bundle", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:16:45.373156366Z", "CreatedBy": "/bin/sh -c #(nop)  ENV PATH=/usr/local/bundle/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "Id": "<missing>", "Size": 0, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:16:46.102599387Z", "CreatedBy": "/bin/sh -c mkdir -p \"$GEM_HOME\" && chmod 777 \"$GEM_HOME\"", "Id": "sha256:a7a2582e3a281ffbecf7e71382dee50443dfc4753756680f35b5be99910c6443", "Size": 143, "Tags": []}, {"Comment": "", "Created": "2020-08-05T07:16:46.286957881Z", "CreatedBy": "/bin/sh -c #(nop)  CMD [\"irb\"]", "Id": "<missing>", "Size": 0, "Tags": []}]
Metadata Type: docker_history
```

### CODE REVIEW

File: anchorecli/cli/image.py [hunk 1]
```diff
@@ -291,17 +291,19 @@ def query_metadata(input_image, metadata_type):
...
-                        o = json.loads(anchorecli.cli.utils.format_output(config, 'image_metadata', {'query_type':metadata_type}, ret['payload']))
-                        print(json.dumps(o, indent=4))
-                    else:
-                        print(anchorecli.cli.utils.format_output(config, 'image_metadata', {'query_type':metadata_type}, ret['payload']))
+                    print(anchorecli.cli.utils.format_output(config,
+                                                             'image_metadata',
+                                                             {'query_type': metadata_type},
+                                                             ret['payload']))
                 else:
-                    raise Exception (json.dumps(ret['error'], indent=4))
```
Comment (alfredodeza) [inline anchorecli/cli/image.py:304]: thank you

File: anchorecli/cli/utils.py [hunk 2]
```diff
@@ -335,7 +335,12 @@ def format_output(config, op, params, payload):
                         t.add_row(row)
                     obuf = obuf + t.get_string(sortby='Package')
                 elif params['query_type'] in ['manifest', 'dockerfile', 'docker_history']:
-                    obuf = format_content_query(payload)
+                    if op == 'image_content':
+                        obuf = format_content_query(payload)
+                    else:
+                        # Metadata Query. Note: The design of this whole method is bad, just doing the change in place
+                        # to reduce changes for now, but should refactor this thing later
```
Comment (alfredodeza) [inline anchorecli/cli/utils.py:342]: Agreed on this!

File: tests/unit/cli/test_utils.py [hunk 3]
```diff
@@ -153,6 +153,33 @@ def test_content_gets_decoded(self, content):
...
+    def test_metadata_cannot_be_decoded(self):
+        payload = {'metadata': b'\t23hsdf'}
+        result = utils.format_metadata_query(payload)
+        assert result == ''
+
+    @pytest.mark.parametrize('metadata', [_manifest_metadata, _docker_history_metadata])
+    def test_content_gets_decoded(self, metadata):
+        payload = {'metadata': metadata}
+        result = utils.format_metadata_query(payload)
+        assert result != ''
```
Comment (alfredodeza) [inline tests/unit/cli/test_utils.py:180]: I get that the result will always be different, but you can still try to assert something that you know is constant. In this case, this test would pass if `result is None` for example, and that would go undetected.

You still have access to the incoming variable so the test could ensure that is included:

```suggestion
        assert metadata in result
```

File: tests/unit/cli/test_utils.py [hunk 4]
```diff
@@ -153,6 +153,33 @@ def test_content_gets_decoded(self, content):
         assert result == 'FROM centos7'
 
 
+class TestFormatMetadataQuery:
```
Comment (alfredodeza) [inline tests/unit/cli/test_utils.py:156]: there are no tests for `metadata_type` or `imageDigest`

File: tests/unit/cli/test_utils.py (hunk 3 above)
Comment (dakaneye) [reply tests/unit/cli/test_utils.py:180]: I slightly modified this to make sure the string is formatted, but don't want to mess with b64 decoding during the test, unless you disagree

File: tests/unit/cli/test_utils.py (hunk 4 above)
Comment (dakaneye) [reply tests/unit/cli/test_utils.py:156]: added

**Pull request merged**